src/xbot/common/subsystems/PoseSubsystem.java

```diff
@@ -0,0 +1,184 @@
+package xbot.common.subsystems;
+
+import org.apache.log4j.Logger;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.sensors.NavImu.ImuType;
+import xbot.common.controls.sensors.XGyro;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.XYPair;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+@Singleton
+public class PoseSubsystem extends BaseSubsystem {
```

Comment:
After an offline discussion, we decided that since the regular update requires distance inputs, that any "telemetry updates" will happen in the DriveSubsystem (which owns the distance sensors)